modules/core/src/lib/effect-manager.js

```diff
@@ -25,25 +27,28 @@ export default class EffectManager {
     return redraw;
   }
 
-  setEffects(effects = []) {
-    this.effects = effects;
-  }
-
   getEffects() {
-    return this.effects;
+    let effects = this.effects;
+    if (this.needApplyDefaultLighting) {
+      effects = this.effects.slice();
+      effects.push(this.defaultLightingEffect);
+    }
+    return effects;
   }
 
   // Private
-  applyDefaultLightingEffect() {
+  setEffects(effects = []) {
+    this.effects = effects;
+  }
+
+  checkLightingEffect() {
     let hasEffect = false;
     for (const effect of this.effects) {
       if (effect instanceof LightingEffect) {
         hasEffect = true;
         break;
       }
     }
-    if (!hasEffect) {
-      this.effects.push(new LightingEffect());
-    }
+    this.needApplyDefaultLighting = !hasEffect;
```

Comment:
Set this to true by default and set to false in the `if (effect instanceof LightingEffect) {` block?

Reply:
good suggestion. Thanks!